### DISCUSSION TITLE
First Find

### DISCUSSION THREAD

**Discussion by tiju86** [apache/couchdb#4727] [Q&A]
Hello,

With CouchDB 3.2.1, I am experiencing slowness when using the find function.
During the initial loading, the find operation takes a long time: approximately 15 seconds for 15 documents.
On the second loading, it is instantaneous.

Is this normal? (Cache?)
How can I improve this loading time or add some debugging?

My code:

```
find({
    selector: {
        type_document: 'article',
    }
});
```

thanks !

**Comment by 28101991SUNNY** [comment]
"Make sure you've set up the right indexes for the fields you're searching. If you're frequently looking for documents with the 'type_document' field set to 'article', it's a good idea to create an index for it."
"CouchDB stores query results in its cache, which might be why subsequent queries are quicker. However, it's unusual for the initial query to take 15 seconds for just 15 documents."
"Monitor the server's system resources like CPU, memory, and disk I/O during the initial load. Slow loading times may be due to resource limitations on the server."
"Double-check your query to ensure it's optimized. Avoid complex or unnecessary queries that could potentially slow down the database."
